### DISCUSSION TITLE
Unable to navigate the tour with kayboard

### DISCUSSION THREAD

**Discussion by jiyang-sas** [gilbarbara/react-joyride#650] [Q&A] [answered]
Hey, I am having problem in navigating the tour using keyboard. Once the tour starts, the only key it will  respond to is the ESC key, Any other keys such as the tab key has no effect on the tour tooptip component. It looks like the focus is not on the tour component. Any sugestions?  Thanks a lot for the great libarary, we really love it.

**Comment by jiyang-sas** [comment]
It turns out that I was using react-joyride 2.2.1,  I was calling backProps.onClick() inside my button press handler. Somehow this call has no effect in 2.2.1. If we switch to 2.3.0, it works.  

I had to use 2.2.1 for now because we are still using reqact 16.

**Reply by gilbarbara** [reply]
if you are using a custom component you'll need to implement the keyboard navigation by yourself.
Anyway, 2.3.0 is compatible with React 15/16/17...

**Comment by gilbarbara** [comment, marked as answer]
Anyway, I don't see how the custom component handlers have anything to do with keyboard navigation.

**Reply by jiyang-sas** [reply]
While I was using react-joyride 2.3.0, I got a compiler error: 

../node_modules/react-joyride/types/index.d.ts:150:16 - error TS2724: 'React' has no exported member named 'RefCallback'. Did you mean 'useCallback'?

This line of the code is defined in 2.3.0 of TooltipRenderProps:
  .....
  tooltipProps: {
    'aria-modal': boolean;
    ref: React.RefCallback<HTMLElement>;
    role: string;
  };

We are using "react": "^16.12.0", If I switch to react-joyride 2.2.1, the error goes away.   Any idea? Thanks!

**Reply by gilbarbara** [reply]
Update the @types/react package?

**Reply by jiyang-sas** [reply]
Thanks a  lot!
